services/remotecluster/ping.go

```diff
@@ -97,6 +97,12 @@ func (rcs *Service) pingRemote(rc *model.RemoteCluster) error {
 		return err
 	}
 
+	if metrics := rcs.server.GetMetrics(); metrics != nil {
+		sentAt := time.Unix(0, ping.SentAt*int64(time.Millisecond))
+		elapsed := time.Since(sentAt).Seconds()
+		metrics.ObserveRemoteClusterPingDuration(rc.RemoteId, elapsed)
+	}
```

Comment:
Here we are storing the actual round-trip time for the ping message which doesn't suffer from clock skew.
If you want, we can add an additional metric for `ping.RecvAt - ping.SentAt`.

Reply:
I included the extra fields in the messaging in case we needed to detect clock skew.  We know how long the round-trip was, so the difference between that and RecAt-SentAt should be clock skew.

If the clocks are way off, then we'll get messages in different orders at each site.  If you could add a clock-skew metric that would be great.  If not we can add it later.

Reply:
To clarify, we would be assuming that each side of the roundtrip takes the same time right? Currently doing some research on this, and one way that's suggested in the literature is to query the NTP servers themselves to know the clock skew.

Reply:
Yes, there are certainly more reliable ways to get the actual clock skew. Here we would be doing an approximation with the data we have:

```
RT/2 - (RecvAt-SentAt)
```